core/icons/comment_icon.ts

```diff
@@ -0,0 +1,326 @@
+/**
+ * @license
+ * Copyright 2023 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import {Block} from '../block.js';
+import {BlockSvg} from '../block_svg.js';
+import {COMMENT_TYPE} from './icon_types.js';
+import {Coordinate} from '../utils.js';
+import * as dom from '../utils/dom.js';
+import * as eventUtils from '../events/utils.js';
+import {Icon} from './icon.js';
+import {IHasBubble} from '../interfaces/i_has_bubble.js';
+import {ISerializable} from '../interfaces/i_serializable.js';
+import {Rect} from '../utils/rect.js';
+import * as registry from './registry.js';
+import {Size} from '../utils/size.js';
+import {Svg} from '../utils/svg.js';
+import {TextBubble} from '../bubbles/text_bubble.js';
+import {TextInputBubble} from '../bubbles/textinput_bubble.js';
+import {WorkspaceSvg} from '../workspace_svg.js';
+
+export class CommentIcon extends Icon implements IHasBubble, ISerializable {
+  /** The type string used to identify this icon. */
+  static readonly TYPE = COMMENT_TYPE;
+
+  /**
+   * The weight this icon has relative to other icons. Icons with more positive
+   * weight values are rendered farther toward the end of the block.
+   */
+  static readonly WEIGHT = 3;
+
+  /** The size of this icon in workspace-scale units. */
+  private readonly SIZE = 17;
```

Comment:
Why is the weight static but the size isn't? They seem pretty similar in concept to me.

Reply:
See discussion here: https://github.com/google/blockly/pull/7115#discussion_r1205992681As per Christopher's suggestion, I'm going to make this module-const.